pkg/kubernetes/pod/v1alpha1/kubernetes.go

```diff
@@ -75,6 +88,11 @@ type KubeClientBuildOption func(*KubeClient)
 // withDefaults sets the default options
 // of KubeClient instance
 func (k *KubeClient) withDefaults() {
+	if k.getKubeConfig == nil {
+		k.getKubeConfig = func() (config *rest.Config, err error) {
+			return client.New().Config()
```

Comment:
How do we handle the case where kubeconfigpath is provided.

Reply:
we might need to do something similar to getClientset & getClientsetForPath & getClientsetForPathOrDirect

Reply:
Done